target/linux/ath79/patches-6.6/820-mfd-syscon-remove-reset-control-for-syscon-devices.patch

```diff
@@ -0,0 +1,67 @@
+From: Shiji Yang <yangshiji66@outlook.com>
+Date: Wed, 13 Mar 2024 22:36:31 +0800
+Subject: [PATCH] mfd: syscon: remove reset control for syscon devices
```

Comment:
Could we consider a different approach to this 6.3 mfd syscon reset issue here? If we one day need to use a driver that relies on this upstream behaviour, we would need to come back to this.
Best wait to see what committers think before changing, I don't want to run you in circles.

Perhaps we:
- add a gpio property (or new `syscon-no-reset`? compatible) to tell syscon not to claim reset?
  ```c
   if (!of_property_read_bool(np, "syscon-no-reset")) {
   	reset = of_reset_control_get_optional_exclusive(np, NULL);
  ```
- work out how we go about having the mfd (sub)device driver (ag71xx here) claim the reset instead?

@dangowrt made some comments upstream, maybe they might have ideas here?

Cheers

Reply:
> If we one day need to use a driver that relies on this upstream behaviour, we would need to come back to this.

I believe this will never happen, the ath79 target is already very old.

> * add a gpio property (or new `syscon-no-reset`? compatible) to tell syscon not to claim reset?

A good idea. If more reviewers request a new property here, I'll add it. Thanks.

> * work out how we go about having the mfd (sub)device driver (ag71xx here) claim the reset instead?

At least the `shared` reset controller doesn't work. I've already tried it. I have no idea now.

Anyway, this commit looks strange to me. It didn't export the requested reset controller at all.
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit?id=7d1e3bd94828ad9fc86f55253cd6fec8edd65394